api/chalicelib/core/projects.py

```diff
@@ -62,21 +63,20 @@ def get_projects(tenant_id: int, gdpr: bool = False, recorded: bool = False):
                                       (SELECT MIN(sessions.start_ts)
                                        FROM public.sessions
                                        WHERE sessions.project_id = s.project_id
-                                         AND sessions.start_ts >= (EXTRACT(EPOCH 
+                                         AND sessions.start_ts >= (EXTRACT(EPOCH
                                                         FROM COALESCE(s.sessions_last_check_at, s.created_at)) * 1000-%(check_delta)s)
                                          AND sessions.start_ts <= %(now)s
                                        )) AS first_recorded"""
 
-        query = cur.mogrify(f"""{"SELECT *, first_recorded IS NOT NULL AS recorded FROM (" if recorded else ""}
+        rows = await txn.execute(f"""{"SELECT *, first_recorded IS NOT NULL AS recorded FROM (" if recorded else ""}
```

Comment:
is this `txn.execute` or `cur.execute` ?

Reply:
It is `txn` for transaction.

Reply:
I do not use cursors in cases like this. Do you recommend the use of cursors? when? and why?

Reply:
I use cursors only when I need to iterate over large set. WDYT?